### ISSUE TITLE
'Windowed' Sidebar widgets need to remember their dimensions and open/close state

### ISSUE THREAD

**Issue by op2786** [Vector35/binaryninja-api#6067]
**Version and Platform (required):**
- Binary Ninja Version: 4.2.6330-dev, ce297927
- OS: MacOS
- OS Version: 15.0
- CPU Architecture: arm64

**Bug Description:**
I believe it should only open when a user ask for it.

**Steps To Reproduce:**
1. Right click on the cross references sidebar icon and select Window mode
2. Open a binary 
3. See the cross references window is opened in background

**Comment by psifertex** [comment]
This is true of all panels that have been turned into "Window" style windows, not just xrefs. The idea is that they're generally available so you can lay them out however you want.

The problem is that if you want xrefs to behave like a modal dialog then window is currently the closest option and it has this negative side-effect.

The solution is either:
1) Implement a new "popup" mode that is different than window
or
2) Fix the save/restore code so that the popup remembers it wasn't open and stays closed, and also change the windowed behavior of xref so that the xref window closes itself when you navigate a new location.

The second approach is our current plan for solving this for now and is actually already tracked under https://github.com/Vector35/binaryninja-api/issues/6026

In other words, if the window wasn't open before, it shouldn't be open again.

**Comment by plafosse** [comment]
Fixed in 4.2.6383
